flac-1.3.0 pkgconfig files do use -I<includedir> and not -I<includedir>/FLAC.

--- modules/codec/flac.c.orig	2013-06-04 11:22:58.984909594 +0200
+++ modules/codec/flac.c	2013-06-04 11:23:09.463977489 +0200
@@ -38,8 +38,8 @@
 #include <vlc_codec.h>
 #include <vlc_aout.h>
 
-#include <stream_decoder.h>
-#include <stream_encoder.h>
+#include <FLAC/stream_decoder.h>
+#include <FLAC/stream_encoder.h>
 
 #include <vlc_block_helper.h>
 #include <vlc_bits.h>
